### PR TITLE
✨ [RUMF-921] differentiate handled and unhandled errors

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

In the context of RUM, an error is considered handled as long as it has been reported manually in the code (e.g console.error, rumAgent.addError). Errors collected through global error/exception listeners are considered unhandled.
This new tag differentiate handled and unhandled errors.

## Changes

Add `handling` flag to all errors reported by RUM except Network and Agent errors

## Testing

Unit, Locally

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by BenoitZugmeyer — COMMENTED** [review]
LGTM so far :)

File: packages/rum-core/src/rumEvent.types.ts [hunk 1]
```diff
@@ -131,6 +131,10 @@ export type RumErrorEvent = CommonProperties & {
      * The type of the error
      */
     readonly type?: string
+    /**
+     * Whether the error has been handled manually in the source code or not
+     */
+    readonly handling?: 'handled' | 'unhandled'
```
Comment (bcaudan) [inline packages/rum-core/src/rumEvent.types.ts:137]: should we also add this field to logs error?

Comment (amortemousque) [reply packages/rum-core/src/rumEvent.types.ts:137]: I've asked Maxime and for him, it's good enough for now.

**Review by bcaudan — APPROVED** [review]
LGTM

**Pull request merged**